### PR TITLE
Use remote data in question list and question view

### PR DESCRIPTION
**Pull request by stephenjust**
This PR enables the use of remote data in question list and question view.

There was a little bit of refactoring in tasks and DataManager. @linster and @RyanKusters please check this.

### CODE REVIEW

**Comment by RyanKusters** [comment]
Looks good, put it in.

File: QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/MainActivity.java [hunk 1]
```diff
@@ -138,10 +154,7 @@ public int compare(Question q1, Question q2) {
 			}
 		case 3:{
 			final List<UUID> favQ = cd.getFavoriteQuestions();
-			for (UUID q : favQ){
```
Comment (linster) [inline QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/MainActivity.java:141]: Not sure why this was removed?

File: QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/QuestionActivity.java [hunk 2]
```diff
@@ -148,19 +128,8 @@ public void onItemClick(AdapterView<?> parent, View view, int position,
...
-			//Set the star to be highlighted on create.
-			ImageButton Favbutton = (ImageButton)findViewById(R.id.question_view_fav_button);
-			Favbutton.setImageResource(R.drawable.ic_fav_highlighted);	
-		} else {
-			//Question is not favorited.
-			ImageButton Favbutton = (ImageButton)findViewById(R.id.question_view_fav_button);
-			Favbutton.setImageResource(R.drawable.ic_fav_reg);
-		}		
+		// Get question data
+		dataManager.getQuestion(questionId, new QuestionUpdateCallback());
```
Comment (linster) [inline QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/QuestionActivity.java:132]: Very nice simplification.

File: QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/data/DataManager.java [hunk 3]
```diff
@@ -128,29 +127,13 @@ public void addQuestion(Question validQ, Callback<Void> c) {
...
-			//User wants a question from within a thread, or doesn't care about threading
-			//AsyncTasks cannot be nested as they run in one threadpool. Therefore, we must
-			//do something evil
-			return gqt.blockingRun(new UUID[]{id});
+			return task.blockingRun(id);
 		}
-		gqt.setCallBack(new Callback<Question>() {
-			@Override
-			public void run(Question q) {
-				recentVisit.add(q.getId());
```
Comment (linster) [inline QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/data/DataManager.java:143]: According to Stephen, updating the recentVisit list now happens in the GetQuestionTask. Also apparently calling .execute() more than once on an AsyncTask causes an exception.

File: QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/data/threading/AddQuestionTask.java [hunk 4]
```diff
@@ -16,24 +19,22 @@ public AddQuestionTask(Context c) {
...
-		
+		try {
+			// All questions created by the user should be saved locally
+			local.putQuestion(q);
+			local.save();
+		} catch (IOException e) {
+			Log.e("AddQuestionTask", "Failed to create question.");
+		}
+		return null;
 	}
```
Comment (linster) [inline QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/data/threading/AddQuestionTask.java:38]: Nice re-write.

**Comment by linster** [comment]
I have reviewed this pull request and approve this message.

**Pull request merged**